src/PlatformHelper.cs

```diff
@@ -241,15 +243,23 @@ internal static Date ConvertStringToDate(string text)
         /// <param name="text">String to be converted.</param>
         /// <param name="timeOfDay">Time of the day</param>
         /// <returns>Whether the value is a valid time of day</returns>
-        internal static bool TryConvertStringToTimeOfDay(string text, out TimeOfDay timeOfDay)
+        internal static bool TryConvertStringToTimeOfDay(ReadOnlySpan<char> text, out TimeOfDay timeOfDay)
         {
             timeOfDay = default(TimeOfDay);
             if (text == null || !PlatformHelper.TimeOfDayValidator.IsMatch(text))
             {
                 return false;
             }
 
-            return TimeOfDay.TryParse(text, CultureInfo.InvariantCulture, out timeOfDay);
+            TimeSpan time;
+            bool b = TimeSpan.TryParse(text, CultureInfo.InvariantCulture, out time);
+            if (b && time.Ticks >= TimeOfDay.MinTickValue && time.Ticks <= TimeOfDay.MaxTickValue)
+            {
+                timeOfDay = new TimeOfDay(time.Ticks);
+                return true;
+            }

```

Comment:
Why this change? Does the change from string to span change the `TryParse` behaviour?

Reply:
Yes. TimeOfDay.TryParse calls 'TimeSpan.TryParse' and do the if clause.
Since it is using the 'System.String' version, and now we'd call 'ReadOnlySpan<char>' version. I don't want to change the definition of TimeOfDay. So, use the codes directly here. #won't fix